Classes/Base/Context.php

```diff
@@ -222,15 +222,15 @@ protected static function addProcess( \Aimeos\MShop\Context\Item\Iface $context
 			return $fcn( $context );
 		}
 
-		// Reset before child processes are spawned to avoid lost DB connections afterwards (TYPO3 9+ only)
-		$pool = GeneralUtility::makeInstance( 'TYPO3\CMS\Core\Database\ConnectionPool' );
-
-		if( php_sapi_name() === 'cli' && method_exists( $pool, 'resetConnections' ) === true ) {
+		// Reset before child processes are spawned to avoid lost DB connections afterwards (TYPO3 9.4 and above)
+		if( php_sapi_name() === 'cli'
+			&& version_compare( TYPO3_version, '9.4', '>=' ) )
```

Comment:
You should keep the test for `resetConnections()` because it's marked as @internal and can vanish in the future. We've opened a ticket in TYPO3 Forge to make it part of the official interface (hopefully)